pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -947,6 +947,14 @@ void receiveIndividualMessagesFromBatch(MessageMetadata msgMetadata, ByteBuf unc
                     ++skippedMessages;
                     continue;
                 }
+                if (singleMessageMetadataBuilder.getCompactedOut()) {
```

Comment:
can you add a simple test for it?

Reply:
No simple test, but added a test anyhow. Pulled back some extra stuff I had in a later patch too.